.github/workflows/library.yml

```diff
@@ -285,7 +285,7 @@ jobs:
       ## Install with Tests Environment
 
       - name: Install a tests environment
-        if: matrix.task == 'tests' || matrix.task == 'cypress' || matrix.task == 'slow' || matrix.task == 'pyright'
```

Comment:
What was the reason for removing the tests install here?

Reply:
Yeah so it seems like the new poetry will uninstall any extras which aren't specified on the install command... Therefore since tests are also specified to have doctests environment installed, I had to add the `-E tests` down there since the extras installed don't accumulate.

We could leave tests in here but that's an unnecessary step the way its currently setup for poetry 1.2... But I can look closer at the poetry install command there might be a way to keep the behaviour as before.

Reply:
Oh that is painful. A benefit of splitting doctests up from tests is that for the `pyright` run through not as many dependencies needed to be installed, speeding up non-cached CI runs.

...if there isn't an obvious way to fix it, potentially just better to join "tests" and "doctests" into the one install step.

> But I can look closer at the poetry install command there might be a way to keep the behaviour as before.

This would certainly be preferred, but, don't sink too much time into it. If you do opt to change this approach can you leave a "TODO" comment to look into seeing how to make poetry not uninstall unspecified extras.

...maybe this is something where we just wait for the following to land:

https://github.com/python-poetry/poetry/issues/4842

Reply:
Yeah so reading poetry the docs it seems that it's not possible to replicate the old behaviour for that install command. For now I might join the tests and doctests and leave a TODO to address this once groups as extras are supported in poetry.